src/toast/ops/polyfilter.py

```diff
@@ -0,0 +1,607 @@
+    )
+
+    shared_flag_mask = Int(0, help="Bit mask value for optional shared flagging")
+
+    view = Unicode(
+        None, allow_none=True, help="Use this view of the data in all observations"
+    )
+
+    @traitlets.validate("shared_flag_mask")
+    def _check_shared_flag_mask(self, proposal):
+        check = proposal["value"]
+        if check < 0:
+            raise traitlets.TraitError("Shared flag mask should be a positive integer")
+        return check
+
+    @traitlets.validate("det_flag_mask")
+    def _check_det_flag_mask(self, proposal):
+        check = proposal["value"]
+        if check < 0:
+            raise traitlets.TraitError("Det flag mask should be a positive integer")
+        return check
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        return
+
+    @function_timer
+    def _exec(self, data, detectors=None, **kwargs):
+        log = Logger.get()
+
+        if self.pattern is None:
+            pat = None
+        else:
+            pat = re.compile(self.pattern)
+
+        for obs in data.obs:
+            # Get the detectors we are using for this observation
+            dets = obs.select_local_detectors(detectors)
+            if len(dets) == 0:
+                # Nothing to do for this observation
+                continue
+
+            views = obs.view[self.view]
+            for iview, view in enumerate(views):
+                shared_flags = obs.shared[self.shared_flags].data & self.shared_flag_mask
+
+                if self.view is not None:
+                    local_starts = []
+                    local_stops = []
+                    for interval in obs.intervals[self.view]:
+                        local_starts.append(interval.first)
+                        local_stops.append(intervallast)
+                else:
+                    local_starts = [0]
+                    local_stops = [obs.n_local_samples - 1]
+
+                local_starts = np.array(local_starts)
+                local_stops = np.array(local_stops)
+
+                for idet, det in enumerate(dets):
+                    # Test the detector pattern
+                    if pat.match(det) is None:
+                        continue
+
+                    det_flags = obs.detdata[self.det_flags][idet] & self.det_flag_mask
+                    signal = obs.detdata[self.det_data][idet]
+
+                    flags = shared_flags | det_flags
+
+                    filter_polynomial(self.order, flags, [signal], local_starts, local_stops)
+
+                    obs.detdata[self.det_flags][idet][flags] & self.poly_flag_mask
+
+        return
+
+    def _finalize(self, data, **kwargs):
+        return
+
+    def _requires(self):
+        req = {
+            "meta": list(),
+            "shared": [self.shared_flags],
+            "detdata": [self.det_data, self.det_flags],
+            "intervals": [self.view],
+        }
```

Comment:
Same note here about only appending flags / view if they are not None.

Reply:
Done.